src/internal.c

```diff
@@ -699,25 +699,25 @@ int GenerateKey(byte hashId, byte keyId,
     blocks = keySz / digestSz;
     remainder = keySz % digestSz;
 
-    ret = wc_HashInit(&hash, hashId);
+    ret = wc_HashInit(&hash, (enum wc_HashType)hashId);
     if (ret == WS_SUCCESS)
-        ret = wc_HashUpdate(&hash, hashId, kSzFlat, LENGTH_SZ);
+        ret = wc_HashUpdate(&hash, (enum wc_HashType)hashId, kSzFlat, LENGTH_SZ);
     if (ret == WS_SUCCESS && kPad)
-        ret = wc_HashUpdate(&hash, hashId, &pad, 1);
+        ret = wc_HashUpdate(&hash, (enum wc_HashType)hashId, &pad, 1);
     if (ret == WS_SUCCESS)
-        ret = wc_HashUpdate(&hash, hashId, k, kSz);
+        ret = wc_HashUpdate(&hash, (enum wc_HashType)hashId, k, kSz);
     if (ret == WS_SUCCESS)
-        ret = wc_HashUpdate(&hash, hashId, h, hSz);
+        ret = wc_HashUpdate(&hash, (enum wc_HashType)hashId, h, hSz);
     if (ret == WS_SUCCESS)
-        ret = wc_HashUpdate(&hash, hashId, &keyId, sizeof(keyId));
+        ret = wc_HashUpdate(&hash, (enum wc_HashType)hashId, &keyId, sizeof(keyId));
```

Comment:
Some of these lines extend past our coding standard 80-character line limit.  Please adjust, thanks.

Reply:
By introducing shorter length of variable, they should be within 80-character line limit now.